ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -512,6 +512,9 @@ func (ipfs *Connector) ConnectSwarms(ctx context.Context) error {
 
 	for _, id := range ids {
 		ipfsID := id.IPFS
+		if ipfsID == nil {
```

Comment:
should `continue` if `id.Error != ""`, `id.IPFS == nil` (hopefully this should never happen), or `id.IPFS.Error != ""`